geostreams/gltg-bmp
# Update Clowder extractor

This is the first working version of the Clowder Extractor for BMP. The PR restructures the code so the database import script can be used by both the API and the extractor.

## How to test:
The easiest way to test is through docker:
- Set up your environment and prepare the database according to the updated instructions in `README.md`.
- Start all the services by running `docker-compose up`.
- Go to `localhost:8000/clowder` and create an account for the admin user set in `CLOWDER_ADMINS` (default is `admin@example.com`).
- Enable the extractor in Clowder settings.
- Create a dataset in Clowder.
- You can test the extractor by uploading the attached files below to the dataset.
- You can query the database at `localhost:8000/bmp-api/practices`, which should show the entries in `practices` table in the database. 

### Test datasets:
- [test1.xlsx](https://github.com/geostreams/gltg-bmp/files/6728819/test1.xlsx)
- [test2.xlsx](https://github.com/geostreams/gltg-bmp/files/6728822/test2.xlsx)
